tests/addons/environment.js

```diff
@@ -30,13 +29,36 @@ define([
 
     } else {
       this.requests = [];
+      this.responses = [];
       // switch to the fake XHR
       this.xhr = Sinon.useFakeXMLHttpRequest();
       this.xhr.onCreate = function (xhr) {
+        if (self.requests.length < self.responses.length) {
+          var mock = self.responses[self.requests.length];
+          setTimeout(function() {
+            xhr.respond(mock.status, mock.headers, mock.body);
+          }, 0);
+        }
         self.requests.push(xhr);
       };
       // respond calls a fake XHR response using SinonJS
-      this.respond = SinonResponder.makeMockResponder(this.requests);
+      this.respond = function (returnValue, mock) {
+        if (arguments.length < 2) {
+          mock = returnValue;
+          returnValue = null;
+        }
+        if (typeof mock === 'undefined') {
+          console.log('Mock does not exist!');
+        }
+        // this has to be here to work in IE
+        setTimeout(function () {
+          if (self.responses.length < self.requests.length) {
+            self.requests[self.responses.length].respond(mock.status, mock.headers, mock.body);
+          }
+          self.responses.push(mock);
+        }, 0);
```

Comment:
This is similar to before, and handles the case where the XHR is already created when the mock response gets setup.